src/app/content/highlights/components/ShowMyHighlights.tsx

```diff
@@ -86,7 +89,17 @@ class ShowMyHighlights extends Component<ShowMyHighlightsProps, { showGoToTop: b
   }
 
   public componentDidUpdate(prevProps: ShowMyHighlightsProps) {
-    if (this.myHighlightsBodyRef.current) { return; }
+
+    if (!prevProps.summaryIsLoading || !this.myHighlightsBodyRef.current) { return; }
+
+    const {summaryHighlights: prevHighlights} = prevProps;
+    const {summaryHighlights: currHighlights} = this.props;
+
+    if (!prevHighlights || !currHighlights) { return; }
+
+    if (!isEqual(lastProp(prevHighlights), lastProp(currHighlights))) {
+      this.props.services.promiseCollector.add(allImagesLoaded(this.myHighlightsBodyRef.current));
+    }
```

Comment:
I tried something like this. It doesn't work because at the moment this runs, the images are not yet appended to the html, but `summaryHighlights` have already been updated.

Reply:
On the other hand [this](https://github.com/openstax/rex-web/pull/489/commits/ca01867c476cd4e30c76af94892deb9f59df946d) does work (tested with few 34 pages worth of highlights images+maths on slow 3g emulation). I dont think that looping over all images and checking one prop is _that_ bad compared to dealing with issues like ^.

cc. @TomWoodward